src/classes/CON_PreferredPhone_Test.cls

```diff
@@ -0,0 +1,450 @@
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2017
+* @group Contacts
+* @group-content ../../ApexDocContent/Contacts.htm
+* @description Unit tests for Preferred Phone
+*/
+@isTest
+public class CON_PreferredPhone_Test {
+    
+    @testSetup
+    static void dataSetup() {
+        
+        UTIL_CustomSettingsFacade.getSettingsForTests(
+            new Hierarchy_Settings__c(Disable_Preferred_Phone_Enforcement__c = False)
+            );
+        
+        // Simulate a user adding their own handler, should not need User_Managed__c to be true
+        Trigger_Handler__c newHandler = new Trigger_Handler__c(
+            Active__c = true,
+            Asynchronous__c = false,
+            Class__c = 'CON_PreferredPhone_TDTM',
+            Load_Order__c = 0.2,
+            Object__c = 'Contact',
+            Trigger_Action__c = 'BeforeInsert;BeforeUpdate'
+        );
+        insert newHandler;
+    }
+  
+  
+    /* When Preferred Phone is blank and the count of valued phone fields is one
+        update Preferred Phone to label of valued phone and copy the value to 
+        standard Phone.
+    */
+    @isTest 
+    static void testSinglePhoneSmartSet() {
+
+        Contact newCont = new Contact(
+            FirstName = 'JohnnyTest1',
+            LastName = 'JohnnyTest',
+            WorkPhone__c = '1234567890'
+        );
+
+        Test.startTest();
+        insert newCont;
+        Test.stopTest();
+
+        List<Contact> newContAfterList = [SELECT Id, FirstName, LastName, Name, WorkPhone__c , Phone, PreferredPhone__c FROM Contact];
+        Map<String, Contact> contactsByLastName = new Map<String, Contact>();
```

Comment:
Minor, but if we're only inserting one contact we probably don't need to have this map by last name?